vnext/ReactUWP/Modules/AlertModuleUwp.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "AlertModuleUwp.h"
+#include <winrt/Windows.UI.Xaml.Controls.h>
+
+#include <Utils\ValueUtils.h>
+
+#pragma warning(push)
+#pragma warning(disable : 4146)
+#include <cxxreact/JsArgumentHelpers.h>
+#pragma warning(pop)
+namespace winrt {
+using namespace Windows::UI::Xaml;
+using namespace Windows::UI::Xaml::Controls;
+} // namespace winrt
+
+#if _MSC_VER <= 1913
+// VC 19 (2015-2017.6) cannot optimize co_await/cppwinrt usage
+#pragma optimize("", off)
+#endif
+
+namespace react {
+namespace uwp {
+//
+// Alert
+//
+
+const char *AlertModule::name = "RCTAlert";
+
+AlertModule::AlertModule() {}
+
+AlertModule::~AlertModule() = default;
+
+std::string AlertModule::getName() {
+  return name;
+}
+
+std::map<std::string, folly::dynamic> AlertModule::getConstants() {
+  return {};
+}
+
+std::vector<facebook::xplat::module::CxxModule::Method>
+AlertModule::getMethods() {
+  return {
+      Method(
+          "showAlert",
+          [](folly::dynamic args, Callback actionResult) {
+            AlertModule::showAlert(args, actionResult);
+          }),
+
+  };
+}
+
+/*static*/ winrt::fire_and_forget AlertModule::showAlert(folly::dynamic &args, Callback actionResult) {
+
+  std::wstring alertTitle;
+  std::wstring alertContent;
+  std::wstring alertPrimaryButtonText;
+  std::wstring alertSecondaryButtonText;
+  std::wstring alertCloseButtonText;
+
+  for (auto arg : args) {
+  
+    for (auto &pair : arg.items()) {
+      const std::string &propertyName = pair.first.getString();
+      const folly::dynamic &propertyValue = pair.second;
+        
+      if (propertyName == "title") {
+        alertTitle = react::uwp::asHstring(propertyValue);
```

Comment:
>alertTitle = react::uwp::asHstring(propertyValue); [](start = 8, length = 50)

asHstring returns a winrt::hstring, but alertTitle is declared as an std::wstring.  There's an additional conversion happening here that's unnecessary, and could create confusion.

I suggest declaring alertTitle as a winrt::hstring instead of a wstring.

Same comment for the other strings.